### PR TITLE
add support for RC and CUDA event profiling to gds_kernel_latency

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
- renamed old branch (prof-with-events) to test changes
  - aim is to make a few more changes to tests
- can use either env var or exec option to enable profiling gds ops

### CODE REVIEW

File: tests/gds_kernel_latency.c [hunk 1]
```diff
@@ -331,6 +338,22 @@ static struct pingpong_context *pp_init_ctx(struct ibv_device *ib_dev, int size,
...
 		}
+	} else {
+		struct ibv_qp_attr attr = {
+			.qp_state        = IBV_QPS_INIT,
+			.pkey_index      = 0,
+			.port_num        = port,
+			.qp_access_flags = IBV_ACCESS_REMOTE_WRITE | IBV_ACCESS_REMOTE_READ | IBV_ACCESS_LOCAL_WRITE
+		};
+
+		if (ibv_modify_qp(ctx->qp, &attr,
```
Comment (drossetti) [inline tests/gds_kernel_latency.c]: can't you consolidate the call to ibv_modify_qp() into a single one after the if/else block ?

File: tests/gds_kernel_latency.c [hunk 2]
```diff
@@ -537,12 +588,19 @@ static int pp_post_work(struct pingpong_context *ctx, int n_posts, int rcnt, uin
...
+			} 
                         ret = pp_post_gpu_send(ctx, qpn);
                         if (ret) {
                                 fprintf(stderr, "ERROR: can't post GPU send\n");
                                 i = -ret;
                                 break;
                         }
+			if (gds_enable_event_prof) {
+				cudaEventRecord(stop_time[event_idx], gpu_stream);
+				event_idx++;
```
Comment (drossetti) [inline tests/gds_kernel_latency.c]: you should clamp the value here to MAX_EVENTS

File: tests/gds_kernel_latency.c [hunk 3]
```diff
@@ -730,11 +792,22 @@ int main(int argc, char *argv[])
 			peersync = !peersync;
                         printf("INFO: switching PeerSync %s\n", peersync?"ON":"OFF");
 			break;
+			
+		case 'K':
+			gds_qpt = (int) strtol(optarg, NULL, 0);
```
Comment (drossetti) [inline tests/gds_kernel_latency.c]: why not a new scope and defining a local variable ?

File: tests/gds_kernel_latency.c [hunk 4]
```diff
@@ -730,11 +792,22 @@ int main(int argc, char *argv[])
...
+                        printf("INFO: GDS_QPT %s\n", (gds_qpt == 1) ? "UD":"RC");
+			gds_qpt = (gds_qpt == 1) ? IBV_QPT_UD : IBV_QPT_RC;
```
Comment (drossetti) [inline tests/gds_kernel_latency.c]: can you return an error if gds_qpt is != (1,2) ?

**Comment by Akshay-Venkatesh** [comment]
@drossetti incorporated your suggestions, Davide.

**Pull request merged**